main/lsp/watchman/WatchmanProcess.cc

```diff
@@ -38,19 +38,22 @@ void WatchmanProcess::start() {
         // laptops have 4.9.0. Thus, we use [ "anyof", [ "suffix", "suffix1" ], [ "suffix", "suffix2" ], ... ].
         // Note 2: `empty_on_fresh_instance` prevents Watchman from sending entire contents of folder if this
         // subscription starts the daemon / causes the daemon to watch this folder for the first time.
-        string subscribeCommand = fmt::format("[\"subscribe\", \"{}\", \"{}\", {{\n"
-                                              "  \"expression\": [\"allof\", "
-                                              "    [\"type\", \"f\"],\n"
-                                              "    [\"anyof\", {}]"
-                                              "  ],\n"
-                                              "  \"defer_vcs\": false,\n"
-                                              "  \"fields\": [\"name\"],\n"
-                                              "  \"empty_on_fresh_instance\": true\n"
-                                              "}}]\n",
-                                              workSpace, subscriptionName,
-                                              fmt::map_join(extensions, ",", [](const std::string &ext) -> string {
-                                                  return fmt::format("[\"suffix\", \"{}\"]", ext);
-                                              }));
+        string subscribeCommand = fmt::format(
+            "[\"subscribe\", \"{}\", \"{}\", {{\n"
+            "  \"expression\": [\"allof\", "
+            "    [\"type\", \"f\"],\n"
+            "    [\"anyof\", {}],\n"
+            "    [\"not\", [\"match\", \"**/.~tmp~/**\", \"wholename\", {{\"includedotfiles\": true}}]]\n" // Exclude
+                                                                                                           // rsync
+                                                                                                           // tmpfiles
```

Comment:
:shakes-fist: clang-format

Do you want to try seeing if clang-format is happier with this:

```suggestion
            // Exclude rsync tmpfiles
            "    [\"not\", [\"match\", \"**/.~tmp~/**\", \"wholename\", {{\"includedotfiles\": true}}]]\n"
```

Reply:
Sure!

Reply:
That works much better, thanks.